sdk/webpubsub/tests.yml

```
@@ -0,0 +1,20 @@
+trigger: none
+
+stages:
+  - template: /eng/pipelines/templates/stages/archetype-sdk-tests.yml
+    parameters:
+      ServiceDirectory: webpubsub
+      Artifacts:
+        - name: azure-messaging-webpubsub
+          groupId: com.azure
+          safeName: azuremessagingwebpubsub
+        - name: azure-messaging-webpubsub-client
+          groupId: com.azure
+          safeName: azuremessagingwebpubsubclient
+      TimeoutInMinutes: 60
+      SupportedClouds: 'Public,UsGov,China'
+      EnvVars:
+        AZURE_LOG_LEVEL: 2
+        AZURE_CLIENT_ID: $(aad-azure-sdk-test-client-id)
+        AZURE_CLIENT_SECRET: $(aad-azure-sdk-test-client-secret)
+        AZURE_TENANT_ID: $(aad-azure-sdk-test-tenant-id)
```

Please change this to the config below, which reduces our dependency on secrets and also supports sovereign cloud testing should that ever be required:

```suggestion
        AZURE_CLIENT_ID: $(WEBPUBSUB_CLIENT_ID)
        AZURE_CLIENT_SECRET: $(WEBPUBSUB_CLIENT_SECRET)
        AZURE_TENANT_ID: $(WEBPUBSUB_TENANT_ID)
        AZURE_SUBSCRIPTION_ID: $(WEBPUBSUB_SUBSCRIPTION_ID)
```

---

@benbp should we be updating this everywhere in out tests as I believe all deployments create these AAD credential variables?

---

@alzimmermsft yes `$(aad-azure-sdk-test-client-id)` is a deprecated secret. What you can also do is update your test base classes that look up these variables to look for either AZURE_* or <service prefix>_*, the latter which you can get from the `AZURE_SERVICE_DIRECTORY` environment variable.